convert-macros/src/lib.rs

```diff
@@ -0,0 +1,105 @@
+use std::iter;
+
+use manyhow::{ensure, manyhow, Result};
+use proc_macro_crate::{crate_name, FoundCrate};
+use quote::{format_ident, quote, ToTokens};
+use syn::{parse_quote, Attribute, DeriveInput, Path};
+
+/// Tries to resolve the path to `extism_convert` dynamically, falling back to feature flags when unsuccessful.
+fn convert_path() -> Path {
+    match crate_name("extism") {
+        Ok(FoundCrate::Name(name)) => {
+            let ident = format_ident!("{name}");
+            parse_quote!(::#ident::convert)
+        }
+        Ok(FoundCrate::Itself) => parse_quote!(crate::core),
```

Comment:
This line was giving me some trouble, but seems to work if I remove `core`:
```suggestion
        Ok(FoundCrate::Itself) => parse_quote!(crate),
```

Reply:
oh sorry that was a copy paste :D